OpenRA.Mods.Common/Traits/Attack/AttackTurreted.cs

```diff
@@ -33,12 +33,15 @@ public AttackTurreted(Actor self, AttackTurretedInfo info)
 		protected override bool CanAttack(Actor self, Target target)
 		{
 			if (!base.CanAttack(self, target))
+			{
+				self.CancelActivity();
 				return false;
+			}
 
 			var canAttack = false;
 			foreach (var t in turrets)
 				if (t.FaceTarget(self, target))
-					canAttack = true;
+					return true;
 
 			return canAttack;
```

Comment:
Since you return true in the loop, you can return false here and do away with the local variable.